RestaurantsNearMe/Presentation/RestaurantListViewModel/RestaurantListViewModel.swift

```diff
@@ -40,11 +53,18 @@ class RestaurantListViewModel: ObservableObject {
 
     restaurantsListProvider.restaurants
       .receive(on: DispatchQueue.main)
-      .assign(to: &$restaurants)
+      .sink { [weak self] in
+        guard let self = self else { return }
+        self.restaurants += $0
+        self.itemsLoadedCount = self.restaurants.count
```

Comment:
the updated results append to our current list.

I was looking more into caching as well, but from what i've read is a lot of this is handled by URLSession already under the hood. Meaning that when we're offline, the result to a *exact* API call is stored meaning unless we need some sort of lifetime persistence, nothing explicitly needs to be done. If that doesn't match best practices i'd love to learn more, but didn't see many resources online.